### PR TITLE
lnd: fix pprof server redirects

### PR DESCRIPTION
**Pull request by yyforyongyu**
Looks like it's causing too many redirects error for some machines.

### CODE REVIEW

**Review by guggero — APPROVED** [review]
tACK, LGTM :tada:

File: lnd.go [hunk 1]
```diff
@@ -187,16 +187,15 @@ func Main(cfg *Config, lisCfg ListenerCfg, implCfg *ImplementationCfg,
 	// Enable http profiling server if requested.
 	if cfg.Profile != "" {
 		// Create the http handler.
-		profileRedirect := http.RedirectHandler(
-			"/debug/pprof", http.StatusSeeOther,
-		)
-		http.Handle("/", profileRedirect)
+		pprofMux := http.DefaultServeMux
+		http.DefaultServeMux = http.NewServeMux()
```
Comment (positiveblue) [inline lnd.go:191]: I am not sure if there is any difference between 

```
pprofMux := http.DefaultServeMux
http.DefaultServeMux = http.NewServeMux()

// or
http.DefaultServeMux = http.NewServeMux()
pprofMux := http.DefaultServeMux
```

I guess we are doing it to isolate the pprof endpoints from other traffic without creating a specific mux for it?

**Review by positiveblue — APPROVED** [review]
tACK :tada: 

I think another way to do it would be to create a specific mux for the pprof calls and register them one by one

```
pprofMux := http.NewServeMux()

pprofMux.Handle("/debug/pprof/", http.HandlerFunc(pprof.Index))
pprofMux.Handle("/debug/pprof/cmdline", pprof.Handler("cmdline"))
pprofMux.Handle("/debug/pprof/profile", pprof.Handler("profile"))
pprofMux.Handle("/debug/pprof/symbol", pprof.Handler("symbol"))
pprofMux.Handle("/debug/pprof/trace", pprof.Handler("trace"))
...
```

I am happy with this solution though :100:

File: lnd.go (hunk 1 above)
Comment (yyforyongyu) [reply lnd.go:191]: yea i have very little knowledge in `http` package. Based on the code I think `pprofMux` is a copy of `http.DefaultServerMux`, and we then overwrite `http.DefaultServerMux` with a new mux. Also reversing the order won't work.

**Comment by yyforyongyu** [comment]
sorry guys you may have to ack it again @positiveblue @guggero . So I think it's better to keep the redirecting, thus we only need to visit `127.0.0.1:6060` to get the profile. Decided to keep this behavior in case other services rely on this redirection.

As for why it was working prior to #7614, I think it's because we used `&http.Serve.ListenAndServe` instead of `http.ListenAndServe`.

File: lnd.go (hunk 1 above)
Comment (guggero) [reply lnd.go:191]: Yeah, good point. I don't think we should overwrite the default serve mux.
Can't we just do this very explicitly?
Here's the diff I have in mind:
[lnd.diff.txt](https://github.com/lightningnetwork/lnd/files/11815202/lnd.diff.txt)

Comment (yyforyongyu) [reply lnd.go]: Nice applied the diff!

**Review by positiveblue — APPROVED** [review]
re approved :+1:

We can add more endpoints (gorutines, etc) if we need them

**Review by guggero — APPROVED** [review]
re-tACK :tada:

**Pull request merged**